### ISSUE TITLE
#1 - 61 kg issue

### ISSUE THREAD

**Issue by theswix** [unitedworldwrestling/arena-public#28]
61 kg. That’s where the first and major problem happened… Take a look at very last wrestler that has no ranking (#7 in the bracket). It is perfectly correct since he didn’t show up at the first weigh-in. 

But in this case, we ran into really strange and serious problem that required us immediate actions to be taken. 
The following had happened: we created matches and when the match started, called him (#7) to mat, he didn’t show up and we have set VFO for that particular match for his opponent (match #236). 

The 1/16 kept going fine but when it was already time to print out the protocols for 1/8 we noticed that match #249 is OK, but match #250 is NOT: wrestler #5 was in protocol, but his opponent (#8 who won in the previous round by VFO in match #236) is missing - simply empty field on the screen and in the protocol, but we realized there should already be #8 in that empty field. Unfortunately, we were so stressed that we didn’t take a screenshot :(. Just believe it…

We already couldn’t wait and called Mikhail Pozdeev, the person who used to run the events in Arena in Russia – he offered a trick - (since we only wanted right scoreboard output and did not care about actual results) to cancel the result of the match 236, change its victory type there from VFO to VIN for the same wrestler and refresh the screen and scoreboard client on that mat. It worked – the wrestler #8 showed up in the bracket, protocol and scoreboard.
BUT. At the same time the wrestler #5 made it further to ¼ and even showed up in the bracket (match #257), which is impossible – the 1/8 hadn’t started yet! 

And here we are, standing in the secretariat and in deep silence waiting for the result of matches #149 and #150… Like in bad Hollywood movie, never knowing how Arena will treat the results of these matches – whether Arena overwrites the bracket for match #257 or crashes… :( Luckily, wrestler #5 won by VFA in match #250 and everything went fine further… 

Also it is quite funny that Arena placed #7 without rank - the thing is that we changed his loss from VFO to VIN and in this case he must be ranked according to the points earned which is zero, but the rank itself - 40 should be present. 

Any thoughts on this issue? What have we done wrong? Everything? :)))

Please refer to the attached documents.
[61 kg.pdf](https://github.com/unitedworldwrestling/arena-public/files/6328718/61.kg.pdf)
![61-problem](https://user-images.githubusercontent.com/54967199/115104478-e1ee7c80-9f82-11eb-858c-048cdd97f4fa.jpg)
[61-final bracket.pdf](https://github.com/unitedworldwrestling/arena-public/files/6328740/61-final.bracket.pdf)

**Comment by csabavirag** [comment]
First of all, we are terribly sorry for the inconvenience you had during the event.
Secondly, unfortunately I could not reproduce the problem you mentioned. I tried closing match 236 by VFO first
![image](https://user-images.githubusercontent.com/10976654/115255486-9b3d8580-a12e-11eb-8e18-6471f52eed99.png)
or finishing match 235 first, then closing match 236 by VFO
![image](https://user-images.githubusercontent.com/10976654/115255605-b7412700-a12e-11eb-8772-c76f3620cd8a.png)
As you can see, in both cases the next match was populated successfully. I even tried to re-save match 236 with victory type of VIN, but that neither produced the behaviour you got during your event. Apart from my failure in terms of reproducing the problem, I can believe it happened, just can't understand why. The workflow should not allow us to happen.
Perhaps you could try it again and see if that fails again on your side.

There is an important thing to note, which caused the ranking irregularity. If someone lost a match by VFO, they gets their status set to "Forfeit". This does not get cleared (in the version 1.5.72.1 - but will be different for Injury after #29) unless manually changed. So I loaded your dump and the first thing I realised was:
![image](https://user-images.githubusercontent.com/10976654/115258084-f7a1a480-a130-11eb-962d-54f9afc86508.png)
That's the reason why you got this wrestler at the bottom of the ranking list and without a ranking position. Of course, after clearing the Athlete's status and setting it back to Accepted, he climbed up to the 35th position.

**Comment by theswix** [comment]
@csabavirag thank you very much for your reply and your efforts! I understood the situation with ranking irregularity - have some thoughts on this issue in terms of UI improvement - will start a new ticket for that.

Regarding the main problem here - I will try to reproduce this situation tonight in a clean event and actual dataset from that tournament. I'm already suspecting that that might be a problem in .xlsx import file instead of .xls - but it seems a bit unrealistic to me... Does Arena log anything in MySQL in production (I know it is pretty supid idea, but... some db binary logs maybe...)?

Anyway, let me try. I will report you ASAP. Only a few issues left from that championship ;)

**Comment by theswix** [comment]
Hello, @csabavirag . 

Well, I've tested this situation in clean environment, actual championship dataset and manual draw input for this weight (no other weights were drawed-in since it is a lot of useless routine) and everything works as expected, in usual workflow. 

I also cannot reproduce this situation, unfortunately. 

Attached are perfect brackets with VFO and VIN for that guy... :((( Scoresheets are also fine in both cases.
Believe me, it happened back then - remember that weird screenshot up there in the thread..?

Anyway, thank you a lot for your time and assistance on this matter! I have a few more tickets to open though ;)

[document(4).pdf](https://github.com/unitedworldwrestling/arena-public/files/6347816/document.4.pdf)
[document(5).pdf](https://github.com/unitedworldwrestling/arena-public/files/6347818/document.5.pdf)
